Sources/Diagnose/DiagnoseCommand.swift

```diff
@@ -0,0 +1,132 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+import SKCore
+
+import struct TSCBasic.AbsolutePath
+
+public struct DiagnoseCommand: AsyncParsableCommand {
+  public static var configuration: CommandConfiguration = CommandConfiguration(
+    commandName: "diagnose",
+    abstract: "Reduce sourcekitd crashes",
+    shouldDisplay: false
+  )
+
+  @Option(
+    name: .customLong("request-file"),
+    help:
+      "Path to a sourcekitd request. If not specified, the command will look for crashed sourcekitd requests and have been logged to OSLog"
+  )
+  var sourcekitdRequestPath: String?
+
+  @Option(
+    name: .customLong("os-log-history"),
+    help: "If now request file is passed, how many minutes of OS Log history should be scraped for a crash."
+  )
+  var osLogScrapeDuration: Int = 60
+
+  @Option(
+    name: .customLong("sourcekitd"),
+    help:
+      "Path to sourcekitd.framework/sourcekitd. If not specified, the toolchain is found in the same way that sourcekit-lsp finds it"
```

Comment:
The idea is that you shouldn’t need to pass the path in most cases anyway because we find `sourcekitd` relative to `sourcekit-lsp`. In the cases where you do need to pass it, I think it’s OK if you need to be exact. 

Also: You can point this ti `sourcekitdInProc.framework/sourcekitdInProc`, so it’s not just a matter of appending `sourcekitd` if it’s pointed at a directory. Anyway, if this turns out to be a churn later, we can always make it more lenient.